src/include/units/physical/si/imperial/base/thermodynamic_temperature.h

```diff
@@ -30,19 +32,20 @@ struct degree_fahrenheit : named_scaled_unit<degree_fahrenheit, basic_symbol_tex
 
 namespace detail {
 
-struct zero_fahrenheit_abs_temperature_ : scaled_unit<ratio(27315 * 9 + 5 * 3200, 900), kelvin> {};
+struct zero_fahrenheit_abs_temperature_ : scaled_unit<ratio(27315 * 9 - 5 * 3200, 900), kelvin> {};
 
 }  // namespace detail
 
 struct fahrenheit_temperature_origin : point_origin<fahrenheit_temperature_origin, kelvin> {
   using reference_origin = kelvin_temperature_origin;
```

Comment:
Shouldn't reference_origin be always defined in the terms of `unit::reference`? Doing this otherwise makes interunit conversions between any units much harder, right?

Reply:
Potentially. It has semantics of a relative distance in that dimension (i.e. a quantity), which doesn't care for the unit. So for the implementation is fine either way. I first wanted to get the semantics right. If the code generation is worse, then we could look at this again. However, I doubt it. It comes to play only during conversion between different origins. There it will have to be converted into a common quantity between the source and destination type, which can all be done at compile-time because it's still a constant. Only then will it be added.

Reply:
I think we should make the definition of origin as simple as possible. Otherwise, less experienced users will complain that the library is over-engineered and hard to use (what they claim about Boost:Units right now). It is easier to add more features in the next release if really needed than try to remove them when users will start depending on them (possibly in a bad way).

Reply:
Ok, but what does that mean in this case? The user never sees that implementation detail. And if the user defines his/her own origin, I argue it is easier if any quantity is accepted, rather than artificially requiring a specific unit - the physical meaning is clear in any case.

Reply:
> Ok, but what does that mean in this case?

For example, I will have to progress it through the ISO C++ Committee and those experts are really picky ;-) Also, there will be users that will be defining their own systems and the definition should be as easy as possible. This is why I strived to provide all information about the unit in one line of definition (and not a few functions or type traits hidden behind the macro as other libraries are doing).

Reply:
Ah, I see. Ok, we can make a `related_point_origin` or `derived_point_origin` which does provide the definition in one line, including the offset. There is a catch however: The offset between the Fahrenheit and the Kelvin scale is rational number that cannot be exactly represented by a double. So I would then still prefer to use a rational number specified in terms of the base unit (in this case, it is (27315 * 9 - 5 * 3200)/900 K = 229835/900 K).

Reply:
Sure, we can introduce a custom type for origin definition.